### DISCUSSION TITLE
Update specific traits on page change (add, remove, rename)

### DISCUSSION THREAD

**Discussion by nguyenvoanhkhoa1** [GrapesJS/grapesjs#5449] [Q&A]
I am making a custom trait for the link component, that allows users to select the page. If there is any change related to a page like add, remove, or rename, the custom trait input select does not update the UI at all. I tried to use the query selector to change the input on change but could not select due to the trait input HTML element is not append to the trait manager. I tried `editor.refresh()` too but it is not working (I know that). I have checked and can not find any method to update the trait.

Can someone know and help me with this issue? Thanks a lot.

**Comment by artf** [comment]
You can subscribe to [page events](https://grapesjs.com/docs/api/pages.html#available-events) and update component traits accordingly
